core-library/src/main/java/org/silverpeas/core/contribution/publication/dao/PublicationDAO.java

```diff
@@ -180,70 +185,66 @@ public static int getNbPubByFatherPath(Connection con, NodePK fatherPK,
-    if (statusSubQuery != null) {
-      selectStatement.append(statusSubQuery);
-    }
-    if (checkVisibility) {
-      selectStatement
-          .append(" AND (( ? > sb_publication_publi.pubBeginDate AND ")
-          .append(
-              "? < sb_publication_publi.pubEndDate ) OR ( ? = sb_publication_publi.pubBeginDate AND ")
-          .append(
-              "? < sb_publication_publi.pubEndDate AND ? > sb_publication_publi.pubBeginHour ) ")
-          .append("OR ( ? > sb_publication_publi.pubBeginDate AND ")
-          .append(
-              "? = sb_publication_publi.pubEndDate AND ? < sb_publication_publi.pubEndHour ) OR "
-                  + "( ? = sb_publication_publi.pubBeginDate AND ? = sb_publication_publi.pubEndDate "
-                  + "AND ? > sb_publication_publi.pubBeginHour AND ? < sb_publication_publi.pubEndHour )) ");
+  private static List<String> getRemovedInstanceIdsLinkedToInstance(final Connection con,
+      final DistributionTreeCriteria criteria) throws SQLException {
+    final String instanceId = criteria.getInstanceId();
+    final JdbcSqlQuery query = JdbcSqlQuery
+        .createSelect("distinct sb_publication_publi.instanceId AS instanceId")
+        .from("sb_node_node")
+          .join("sb_publication_publifather")
+          .on("sb_publication_publifather.nodeid = sb_node_node.nodeid")
+          .and("sb_publication_publifather.instanceId = ?", instanceId)
+        .join(SB_PUBLICATION_PUBLI_TABLE)
+          .on("sb_publication_publifather.pubId = sb_publication_publi.pubId");
+    criteria.getStatusSubQuery().ifPresent(query::addSqlPart);
+    if (criteria.visibilityCheckRequired()) {
+      visibleFilter(query, AND_CONJUNCTION, now(), SB_PUBLICATION_PUBLI_TABLE);
     }
-    selectStatement.append(
-        " WHERE sb_node_node.instanceId = ? GROUP BY sb_node_node.nodeId, sb_node_node" +
-            ".nodefatherid");
-
-    try (PreparedStatement prepStmt = con.prepareStatement(selectStatement.toString())) {
-      java.util.Date now = new java.util.Date();
-      String dateNow = formatDate(now);
-      String hourNow = DateUtil.formatTime(now);
-
-      prepStmt.setString(1, instanceId);
-      if (checkVisibility) {
-        prepStmt.setString(2, dateNow);
-        prepStmt.setString(3, dateNow);
-        prepStmt.setString(4, dateNow);
-        prepStmt.setString(5, dateNow);
-        prepStmt.setString(6, hourNow);
-        prepStmt.setString(7, dateNow);
-        prepStmt.setString(8, dateNow);
-        prepStmt.setString(9, hourNow);
-        prepStmt.setString(10, dateNow);
-        prepStmt.setString(11, dateNow);
-        prepStmt.setString(12, hourNow);
-        prepStmt.setString(13, hourNow);
-        prepStmt.setString(14, instanceId);
-      } else {
-        prepStmt.setString(2, instanceId);
-      }
+    final RemovedSpaceAndComponentInstanceChecker checker = RemovedSpaceAndComponentInstanceChecker.create();
```

Comment:
So here, each time you call create() you would get a different object and hence the caches in the checker are unless

Reply:
The checker is not listening to administration modificiations, so the caches included into the checker are useful into the context of a treatment processing.
But after that they potentially obsolete.

Reply:
A checker instance is built each time it is injected into an injection context (either by annotation or programmatically). A cache is useful in the case such as an instance can be used several times in the scope to which the injection context is bound.
In the case of a scope of method where the checker is only used one time, the cache is useless and therefore its implementation carries useless complexity.

However, if the checker instance is used several times in another or several others scopes, yes, the use of a cache is justified.

Reply:
Yes, obviously.